package/gluon-core/luasrc/lib/gluon/upgrade/010-primary-mac

```diff
@@ -12,45 +12,58 @@ local json = require 'jsonc'
 local platform = require 'gluon.platform'
 local util = require 'gluon.util'
 
+local board_data = json.load('/etc/board.json')
+local network_data = (board_data or {}).network
 
-local function sysfs(...)
-	local path = string.format(...)
-	return function()
-		local addr = util.readfile(path)
-		if addr then
-			return util.trim(addr)
-		end
+local function read(...)
+	local addr = util.readfile(string.format(...))
+	if addr then
+		return util.trim(addr)
 	end
 end
 
-local function eth(index)
-	return sysfs('/sys/class/net/eth%d/address', index)
+local function get_netdev_addr(ifname)
+	return read('/sys/class/net/%s/address', ifname)
+end
+
+local function strip_vlan(ifname)
+	return (ifname:gsub('%..*', ''))
```

Comment:
The outter brackets here can be dropped, no?

Reply:
They are intentional, see 0e681d5c37c58fba935fa1f6b99cfc619c0d629c